x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/rank/textsimilarity/ChunkScorerConfigTests.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.rank.textsimilarity;
+
+import org.elasticsearch.common.io.stream.BytesStreamOutput;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.inference.ChunkingSettings;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class ChunkScorerConfigTests extends ESTestCase {
+
+    public void testChunkingSettingsFromMapReturnsNullForNull() {
+        assertNull(ChunkScorerConfig.chunkingSettingsFromMap(null));
+    }
+
+    public void testChunkingSettingsFromMapWithOnlyMaxChunkSize() {
+        Map<String, Object> map = Map.of("max_chunk_size", 500);
+        ChunkingSettings settings = ChunkScorerConfig.chunkingSettingsFromMap(map);
+
+        assertNotNull(settings);
+        assertEquals(500, settings.asMap().get("max_chunk_size"));
+        assertEquals("sentence", settings.asMap().get("strategy"));
+    }
+
+    public void testChunkingSettingsFromMapWithFullSettings() {
+        Map<String, Object> map = Map.of("strategy", "sentence", "max_chunk_size", 200, "sentence_overlap", 1);
+        ChunkingSettings settings = ChunkScorerConfig.chunkingSettingsFromMap(map);
+
+        assertNotNull(settings);
+        assertEquals(200, settings.asMap().get("max_chunk_size"));
+    }
+
+    public void testDefaultChunkingSettingsWithValidSize() {
+        ChunkingSettings settings = ChunkScorerConfig.defaultChunkingSettings(500);
+
+        assertNotNull(settings);
+        assertEquals(500, settings.asMap().get("max_chunk_size"));
+        assertEquals("sentence", settings.asMap().get("strategy"));
+    }
+
+    public void testDefaultChunkingSettingsWithNull() {
```

Comment:
This test should not pass, because the chunk size should be pulled from the inference endpoint!